rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -122,7 +124,9 @@ std::unordered_set<Opcode*> findMergepoints(rir::Code* srcCode) {
     std::unordered_set<Opcode*> mergepoints;
     // Create mergepoints
     for (auto m : incom)
-        if (std::get<1>(m).size() > 1)
+        // The first position must also be considered a mergepoint in case it
+        // has only one incoming (a jump)
+        if (std::get<0>(m) == first || std::get<1>(m).size() > 1)
```

Comment:
It would be really useful if you could also figure out why this caused the vm to grab all the memory insted of just throwing an assert or something else, which is easier to debug. (and fix that too)

Reply:
Let's finish this PR that already made me suffer a lot :). I open #488